src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -2249,7 +2248,7 @@ internal static bool CheckFeatureAvailability(SyntaxTree tree, MessageID feature
             return false;
         }
 
-        private static CSDiagnosticInfo GetFeatureAvailabilityDiagnosticInfo(SyntaxTree tree, MessageID feature)
+        internal static CSDiagnosticInfo GetFeatureAvailabilityDiagnosticInfo(SyntaxTree tree, MessageID feature)
```

Comment:
I'd instead rip out a new helper that just takes a LanguageVersion and feature and produces the diagnostic info. That way you don't need to interrogate a SyntaxTree ever.